lib/pengine/native.c

```diff
@@ -1387,15 +1387,21 @@ pe__rscs_brief_output(pcmk__output_t *out, GListPtr rsc_list, long options, gboo
     GHashTable *rsc_table = crm_str_table_new();
     GHashTable *active_table = g_hash_table_new_full(crm_str_hash, g_str_equal,
                                                      free, destroy_node_table);
-    GHashTableIter hash_iter;
-    char *type = NULL;
-    int *rsc_counter = NULL;
+    GListPtr sorted_rscs;
     int rc = pcmk_rc_no_output;
 
     get_rscs_brief(rsc_list, rsc_table, active_table);
 
-    g_hash_table_iter_init(&hash_iter, rsc_table);
-    while (g_hash_table_iter_next(&hash_iter, (gpointer *)&type, (gpointer *)&rsc_counter)) {
+    /* Make a list of the rsc_table keys so that it can be sorted.  This is to make sure
+     * output order stays consistent between systems.
+     */
+    sorted_rscs = g_hash_table_get_keys(rsc_table);
+    sorted_rscs = g_list_sort(sorted_rscs, (GCompareFunc) strcmp);
```

Comment:
This is fine, because this particular function should never be needed by the scheduler, but FYI for background:

We have a few cases like this already with crm_simulate output for cts-scheduler. In those cases, we don't want to add the sorting overhead unnecessarily for actual scheduler runs, so we use pe_flag_stdout as a condition (see e.g. expand_notification_data()). If we already had data_set here it might be worth it for extra safety but I don't think it's worth the effort of dragging that in just for this.